server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Keyword.scala

```diff
@@ -0,0 +1,111 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ************************************************************** */
+package org.apache.james.jmap.model
+
+import java.util.Optional
+
+import com.google.common.base.{MoreObjects, Objects}
+import com.google.common.collect.{ImmutableBiMap, ImmutableList}
+import com.ibm.icu.text.UnicodeSet
+import javax.mail.Flags
+import org.apache.commons.lang3.StringUtils
+
+object Keyword {
+  private val FLAG_NAME_MIN_LENGTH = 1
+  private val FLAG_NAME_MAX_LENGTH = 255
+  private val FLAG_NAME_PATTERN = new UnicodeSet("[[a-z][A-Z][0-9]]").add('$').add('_').add('-').freeze
+  val DRAFT = Keyword.of("$Draft")
+  val SEEN = Keyword.of("$Seen")
+  val FLAGGED = Keyword.of("$Flagged")
+  val ANSWERED = Keyword.of("$Answered")
+  val DELETED = Keyword.of("$Deleted")
+  val RECENT = Keyword.of("$Recent")
+  val FORWARDED = Keyword.of("$Forwarded")
+  private[model] val FLAG_VALUE = true
+  private val NON_EXPOSED_IMAP_KEYWORDS = ImmutableList.of(Keyword.RECENT, Keyword.DELETED)
+  private val IMAP_SYSTEM_FLAGS = ImmutableBiMap.builder[Flags.Flag, Keyword]
+    .put(Flags.Flag.DRAFT, DRAFT)
+    .put(Flags.Flag.SEEN, SEEN)
+    .put(Flags.Flag.FLAGGED, FLAGGED)
+    .put(Flags.Flag.ANSWERED, ANSWERED)
+    .put(Flags.Flag.RECENT, RECENT)
+    .put(Flags.Flag.DELETED, DELETED)
+    .build
+
+  private val VALIDATION_MESSAGE = """FlagName must not be null or empty, must have length form 1-255,
+      | must not contain characters with hex from '\\u0000' to '\\u00019'
+      | or {'(' ')' '{' ']' '%' '*' '\"' '\\'} """.stripMargin
+
+  def parse(flagName: String): Option[Keyword] = {
+    if (isValid(flagName)) return Some(new Keyword(flagName))
+    throw new IllegalArgumentException(VALIDATION_MESSAGE)
+  }
+
+  def of(flagName: String): Keyword = parse(flagName).get
+
+  def fromFlag(flag: Flags.Flag): Option[Keyword] = Some(IMAP_SYSTEM_FLAGS.get(flag))
+
+  private def isValid(flagName: String): Boolean = {
+    if (StringUtils.isBlank(flagName)) return false
+    if (flagName.length < FLAG_NAME_MIN_LENGTH || flagName.length > FLAG_NAME_MAX_LENGTH) return false
+    if (!FLAG_NAME_PATTERN.containsAll(flagName)) return false
+    true
+  }
+}
+
+class Keyword(val flagName: String) {
+  def getFlagName(): String = flagName
+
+  def isExposedImapKeyword(): Boolean = !Keyword.NON_EXPOSED_IMAP_KEYWORDS.contains(this)
+
+  def isDraft(): Boolean = Keyword.DRAFT == this
+
+  def asSystemFlag(): Optional[Flags.Flag] = Optional.ofNullable(Keyword.IMAP_SYSTEM_FLAGS.inverse.get(this))
+
+  def asFlags(): Flags = asSystemFlag().map(Flag => new Flags(Flag)).orElse(new Flags(flagName))
+
+  override final def equals(other: Any): Boolean = {
+    if (other.isInstanceOf[Keyword]) {
+      val otherKeyword = other.asInstanceOf[Keyword]
+      return Objects.equal(flagName, otherKeyword.flagName)
```

Comment:
don't use return

Reply:
Yes, and we dont need override this kind of method.